train.py

```diff
@@ -0,0 +1,193 @@
+from torch.utils.data import DataLoader
+from pytorch_lightning.callbacks import TQDMProgressBar
+from pytorch_lightning.loggers.wandb import WandbLogger
+
+from config import Config
+from models.utils import Transforms
+from models.reverse_diffusion import Unet
+from models.forward_diffusion import ForwardDiffusion
+
+
+class UnetTrainingWrapper(pl.LightningModule):
+    def __init__(
+        self,
+        model: Unet,
+        forward_diffusion: ForwardDiffusion,
+        lr: float = 1e-3,
+        l2: float = 0.01,
+        loss_type: str = "mse",
+        ckpt_dir: str = "checkpoints",
+        save_every_n_steps: int = 100,
+    ):
+        super().__init__()
+
+        # model
+        self.model = model
+        self.forward_diffusion = forward_diffusion
+        self.timesteps = forward_diffusion.timesteps
+
+        # save hyperparameters (lr, l2, loss_type, ckpt_dir, save_every_n_steps)
+        self.save_hyperparameters(ignore=["model", "forward_diffusion"])
+
+        # get loss function
+        self.loss_function_module = self._loss_function()
+
+    def _loss_function(self) -> nn.Module:
+        if self.hparams["loss_type"] == "mse":
+            loss = nn.MSELoss()
+        elif self.hparams["loss_type"] == "mae":
+            loss = nn.L1Loss()
+        elif self.hparams["loss_type"] == "huber":
+            loss = nn.HuberLoss()
+        else:
+            raise NotImplementedError()
+
+        return loss
+
+    def _save_models(self):
+        # ckpt specifies directory and name of the file is name of the experiment in wandb
+        save_path = f"{self.hparams['ckpt_dir']}/{self.logger.experiment.name}.ckpt"
+        # saving models
+        torch.save({"model": self.model.state_dict(), "forward_diffusion": self.forward_diffusion.state_dict()}, save_path)
+
+    def forward(self, x: torch.Tensor, t: torch.Tensor) -> torch.Tensor:
+        return self.model(x, t)
+
+    def configure_optimizers(self) -> optim.Optimizer:
+        optimizer = optim.AdamW(self.model.parameters(), lr=self.hparams["lr"], weight_decay=self.hparams["l2"])
+
+        return optimizer
+
+    def training_step(self, batch: torch.Tensor, batch_idx: int) -> float:
+        x = batch["x"]
+
+        batch_size = x.shape[0]
+
+        # sample t
+        t = torch.randint(0, self.timesteps, size=(batch_size,), dtype=torch.long, device=self.device)
+
+        # noise batch
+        x_noisy, added_noise = self.forward_diffusion.forward(x, t)
+
+        # get predicted noise
+        predicted_noise = self.model.forward(x_noisy, t)
+
+        # get loss value for batch
+        loss = self.loss_function_module(predicted_noise, added_noise)
+
+        self.log("loss", loss, on_epoch=True)
+
+        if self.global_step % self.hparams["save_every_n_steps"] == 0:
+            self._save_models()
+
+        return loss
+
+
+def preprocess_dataset(filepath: str, batch_size: int, num_workers: int):
+    dataset = load_dataset(filepath)
```

Comment:
```suggestion
def preprocess_dataset(dataset_name: str, batch_size: int, num_workers: int):
    dataset = load_dataset(dataset_name)
```

Looks like it's not a path to file :)